### DISCUSSION TITLE
Multiple compilers with/without viaIR

### DISCUSSION THREAD

**Discussion by fnanni-0** [NomicFoundation/hardhat#3574] [Troubleshooting] [answered]
I'm working with 2 contracts, A.sol and B.sol, both of which inherit from C.sol. Both contracts share the same compiler settings except for viaIR, which should be enabled for A and disabled for B. A.sol won't compile if viaIR is not enabled. This is the compilers settings that I'm using:

```
solidity: {
    compilers: [
      {
        version: "0.8.17",
        settings: {
          optimizer: {
            enabled: true,
            runs: 200,
          },
        },
      },
    ],
    overrides: {
      "contracts/A.sol": {
        version: "0.8.17",
        settings: {
          viaIR: true,
          optimizer: {
            enabled: true,
            runs: 200,
          },
        },
      },
    }
  },
```

However, compilation of A.sol fails with this configuration as if it was compiled with viaIR disabled:

```
CompilerError: Stack too deep. Try compiling with `--via-ir` (cli) or the equivalent `viaIR: true` (standard JSON) while enabling the optimizer. Otherwise, try removing local variables. When compiling inline assembly: Variable dataEnd is 1 slot(s) too deep inside the stack. Stack too deep. Try compiling with `--via-ir` (cli) or the equivalent `viaIR: true` (standard JSON) while enabling the optimizer. Otherwise, try removing local variables.
```

**Comment by fvictorio** [comment]
Hi @fnanni-0, I cannot reproduce this: https://github.com/fvictorio/hardhat-3574

(I'm not inheriting from `C.sol`, just using it, but that shouldn't be relevant)

**Comment by fnanni-0** [comment, marked as answer]
Hi @fvictorio, I just found the problem. I am working with a big repo and missed a test contract that was importing A similar to this https://github.com/fnanni-0/hardhat-3574/blob/main/contracts/D.sol. Simply compiling D with viaIR enabled as well solved the issue.

Thanks a lot!
